src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -255,32 +244,64 @@ private QualifcationConfiguration createQualifcationConfigurationFor(Project pro
             MappingFile = Path.Combine(tmpProjectFolder, "mapping.json"),
             SnapshotFile = project.SnapshotFilePath,
             TempFolder = tmpProjectFolder,
-            BuildingBlocks = mapBuildingBlocks(project.BuildingBlocks, projects),
+            BuildingBlocks = await mapBuildingBlocks(project.BuildingBlocks, projects),
             SimulationParameters = mapSimulationParameters(project.SimulationParameters, projects),
             SimulationPlots = allPlots.ForProject(projectId),
             Inputs = alInputs.ForProject(projectId)
          };
       }
 
-      private string snapshotFileFor(Project project) => absolutePathFrom(_runOptions.ConfigurationFolder, project.Path);
 
-      private BuildingBlockSwap[] mapBuildingBlocks(BuildingBlockRef[] buildingBlocks, IReadOnlyList<Project> projects) =>
-         buildingBlocks?.Select(bb => mapBuildingBlock(bb, projects)).ToArray();
+      private Task<BuildingBlockSwap[]> mapBuildingBlocks(BuildingBlockRef[] buildingBlocks, IReadOnlyList<Project> projects)
+      {
+         if (buildingBlocks == null)
+            return Task.FromResult(Array.Empty<BuildingBlockSwap>());
+
+         return Task.WhenAll(buildingBlocks.Select(bb => mapBuildingBlock(bb, projects)));
+      }
 
-      private BuildingBlockSwap mapBuildingBlock(BuildingBlockRef buildingBlock, IReadOnlyList<Project> projects)
+      private async Task<BuildingBlockSwap> mapBuildingBlock(BuildingBlockRef buildingBlock, IReadOnlyList<Project> projects)
       {
-         var project = projects.FindById(buildingBlock.Project);
-         if (project == null)
-            throw new QualificationRunException(ReferencedProjectNotDefinedInQualificationFile(buildingBlock.Project));
+         string snapshotFilePath;
+         // Using a project reference
+         if (!string.IsNullOrEmpty(buildingBlock.Project))
+         {
+            var project = projects.FindById(buildingBlock.Project);
+            if (project == null)
+               throw new QualificationRunException(ReferencedProjectNotDefinedInQualificationFile(buildingBlock.Project));
+
+            snapshotFilePath = await snapshotFileFullPathFor(project);
+         }
+         else
+            snapshotFilePath = await snapshotFileFullPathFor(buildingBlock.Path);

```

Comment:
Otherwise use project file. IT can be a relative file or absolute url since we are reusing the same method